syncer/object.go

```diff
@@ -53,19 +68,19 @@ func (s *ObjectSyncer) Sync(ctx context.Context) (SyncResult, error) {
 	// don't pass to user error for owner deletion, just don't create the object
 	// nolint: gocritic
 	if err == errOwnerDeleted {
-		log.Info(string(result.Operation), "key", key, "kind", fmt.Sprintf("%T", s.Obj), "error", err)
+		log.Info(string(result.Operation), "key", key, "kind", fmt.Sprintf("%T", objectWithoutSecretData(s.Obj)), "error", err)
 		err = nil
 	} else if err == ErrIgnore {
-		log.V(1).Info("syncer skipped", "key", key, "kind", fmt.Sprintf("%T", s.Obj))
+		log.V(1).Info("syncer skipped", "key", key, "kind", fmt.Sprintf("%T", objectWithoutSecretData(s.Obj)))
 		err = nil
 	} else if err != nil {
 		result.SetEventData(eventWarning, basicEventReason(s.Name, err),
 			fmt.Sprintf("%T %s failed syncing: %s", s.Obj, key, err))
-		log.Error(err, string(result.Operation), "key", key, "kind", fmt.Sprintf("%T", s.Obj), "diff", diff)
+		log.Error(err, string(result.Operation), "key", key, "kind", fmt.Sprintf("%T", objectWithoutSecretData(s.Obj)), "diff", diff)
```

Comment:
It's a good approach, but maybe we can do it somehow more generic.

The idea would be to reduce the leaking of secrets in logging, thus meaning that the entire obfuscation should take place at the logging component level. Only by doing it in this way we can be 100% sure that we don't leak secrets.

Since you can't hijack the logging in Go and implementing a logging component introduces too much overhead and complexity, we can keep it this way.

Just some small improvements:
  - `ToString` instead of `objectWithoutSecretData`
  - `ToString` should return the `fmt.Sprintf("%T")` of the object. If it fails, return empty string.

Of course, we'll need some tests for it, but unittests will do the trick.